solutions/instances/variables.tf

```diff
@@ -0,0 +1,270 @@
+########################################################################################################################
+# Common variables
+########################################################################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The API Key to use for IBM Cloud."
+  sensitive   = true
+}
+
+variable "existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or an existing resource group in which to provision resources to."
+}
+
+variable "region" {
+  description = "Region where resources will be created"
+  type        = string
+  default     = "us-south"
+}
+
+##############################################################################
+# Log Analysis Variables
+##############################################################################
+
+variable "log_analysis_instance_name" {
+  type        = string
+  description = "The name of the IBM Cloud Logging instance to create."
+  default     = "log-analysis"
+}
+
+variable "log_analysis_plan" {
+  type        = string
+  description = "The IBM Cloud Logging plan to provision. Available: lite, 7-day, 14-day, 30-day, hipaa-30-day"
+  default     = "7-day" # Check it whether lite plan will be allowed or not for log Analysis
+
+  validation {
+    condition     = can(regex("^lite$|^7-day$|^14-day$|^30-day$|^hipaa-30-day$", var.log_analysis_plan))
+    error_message = "The log_analysis_plan value must be one of the following: lite, 7-day, 14-day, 30-day, hipaa-30-day."
+  }
+}
+
+variable "log_analysis_service_endpoints" {
+  description = "The type of the service endpoint that will be set for the Log Analysis instance."
+  type        = string
+  default     = "public-and-private" # Change it to private
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.log_analysis_service_endpoints)
+    error_message = "The specified service_endpoints is not a valid selection"
+  }
+}
+
+variable "log_analysis_tags" {
+  type        = list(string)
+  description = "Tags associated with the IBM Cloud Logging instance (Optional, array of strings)."
+  default     = []
+}
+
+variable "enable_archive" {
+  type        = bool
+  description = "Enable archive on log analysis and activity tracker instances"
+  default     = true
+}
+
+variable "archive_api_key" {
+  type        = string
+  description = "Limited IBM Cloud API key for Log Analysis archiving to COS"
```

Comment:
Explain what do you mean by Limited. Also you need to explain that this is optional and if a value not passed the value of the `ibmcloud_api_key` will be used

Reply:
I have taken the description from the terraform-ibm-observability-module so `Limited` was there itself in the variable description. Updated the description accordingly.